pkg/generator/secret_template_reconciler.go

```diff
@@ -75,7 +103,21 @@ func (r *SecretTemplateReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
-	if secretTemplate.DeletionTimestamp != nil {
+	if secretTemplate.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&secretTemplate, finalizerName) {
+			controllerutil.AddFinalizer(&secretTemplate, finalizerName)
+			if err := r.client.Update(ctx, &secretTemplate); err != nil {
```

Comment:
*thought*: Anyway we can do a similar `defer` update statement here or is it messy

Reply:
Mmm maybe, I'm not here it makes so much sense because we only want to call update in certain circumstances. (finalizer added or removed).

Reply:
Makes sense, not blocking for me.